core/include/webview/detail/engine_base.hh

```diff
@@ -23,132 +23,97 @@
  * SOFTWARE.
  */
 
-#ifndef WEBVIEW_DETAIL_ENGINE_BASE_HH
-#define WEBVIEW_DETAIL_ENGINE_BASE_HH
-
+#pragma once
 #if defined(__cplusplus) && !defined(WEBVIEW_HEADER)
 
-#include "../errors.hh"
 #include "../types.h"
 #include "../types.hh"
-#include "json.hh"
 #include "user_script.hh"
-
 #include <atomic>
-#include <functional>
 #include <list>
 #include <map>
-#include <string>
+
+using binding_t = std::function<void(std::string, std::string, void *)>;
+using sync_binding_t = std::function<std::string(std::string)>;
 
 namespace webview {
 namespace detail {
 
+using noresult = webview::noresult;
+
+class binding_ctx_t {
```

Comment:
I un-nested this class from `engine_base`, and everything appears to work fine.
Let me know if there was a specific reason for the nesting.